firmware/application/apps/ble_rx_app.cpp

```diff
@@ -728,9 +720,13 @@ void BLERxView::on_data(BlePacketData* packet) {
 
     // Log at End of Packet.
     if (logger && logging) {
-        logger->log_raw_data(str_console);
+        logger->log_raw_data(str_console + "\r\n");
     }
 
+    usb_serial_thread->serial_str = str_console + "\r\n";
+    usb_serial_thread->str_ready = true;
```

Comment:
I would prefer a message system between the thread and the apps that want to log.
Actually it's done line this in a lot of places, and it's a way to have multiple producers of log lines, and only one consumer (the usb serial thread)

Reply:
Can you point me to, or explain your vision please. Maybe in what you think this should be done. I'm also no opposed to you hacking that in.

Reply:
I mean use a message system to populate strings to send
Like in Scanner i.e the retune messages.
It will have a queue and somewhat thread protection

Reply:
But for a start it's cool :-)

Reply:
No I understand the queue thing but like I'm not sure I see how to implement. Does the app queue up messages?